enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/openshift/ProjectEnricher.java

```diff
@@ -0,0 +1,84 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.enricher.standard.openshift;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Namespace;
+import io.fabric8.kubernetes.api.model.NamespaceSpec;
+import io.fabric8.maven.core.config.PlatformMode;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.MavenEnricherContext;
+import io.fabric8.openshift.api.model.*;
+
+import static io.fabric8.maven.core.util.kubernetes.KubernetesResourceUtil.removeItemFromKubernetesBuilder;
+
+public class ProjectEnricher extends BaseEnricher {
+    static final String ENRICHER_NAME = "fmp-openshift-project";
+
+    public ProjectEnricher(MavenEnricherContext context) {
+        super(context, ENRICHER_NAME);
+    }
+
+    @Override
+    public void create(PlatformMode platformMode, KubernetesListBuilder builder) {
+        if(platformMode == PlatformMode.openshift) {
+            for(HasMetadata item : builder.buildItems()) {
+                if(item instanceof Namespace) {
+                    Project project = convert(item);
+                    removeItemFromKubernetesBuilder(builder, item);
+                    builder.addToProjectItems(project);
+                }
+            }
+        }
+    }
+
+    private Project convert(HasMetadata item) {
+        Namespace namespace = (Namespace) item;
+        namespace.getMetadata();
+
+        ProjectBuilder builder = new ProjectBuilder();
+        builder.withMetadata(namespace.getMetadata());
+
+
+        if (namespace.getSpec() != null) {
+            NamespaceSpec namespaceSpec = namespace.getSpec();
+            ProjectSpec projectSpec = new ProjectSpec();
+            if (namespaceSpec.getFinalizers() != null) {
+                projectSpec.setFinalizers(namespaceSpec.getFinalizers());
+            }
+            namespaceSpec.getAdditionalProperties()
+                    .forEach((k, v) -> projectSpec.setAdditionalProperty(k, v));
+
+            builder.withSpec(projectSpec);
+        }
+
+        if (namespace.getStatus() != null) {
+            ProjectStatus status = new ProjectStatusBuilder()
+                    .withPhase(namespace.getStatus().getPhase()).build();
+
+            namespace.getStatus().getAdditionalProperties()
+                    .forEach((k, v) -> status.setAdditionalProperty(k, v));
+        }
+
+        Project project = builder.build();
+
+        namespace.getAdditionalProperties()
```

Comment:
Could you please tell why is this required?

Reply:
If the namespace resource object has any additional properties, these should also be passed on to the project resource being created. 

You can refer to Project and Namespace resource classes in kubernetes-model

Reply:
Umm,  still not able to understand why we need this. I don't see any additional properties field here:

https://kubernetes.io/docs/tasks/administer-cluster/namespaces-walkthrough/#create-new-namespaces

As far as I know, these additionalProperties are used internally to store additional information. Or they are set if some portions of object were not able to get parsed properly.

Reply:
I don't think we should care about porting them also during conversion.